test/e2e/suite/issuers/acme/certificate/http01.go

```diff
@@ -271,6 +274,62 @@ var _ = framework.CertManagerDescribe("ACME Certificate (HTTP01)", func() {
+
+		switch {
+		case util.HasIngresses(f.KubeClientSet.Discovery(), networkingv1.SchemeGroupVersion.String()):
+			ingClient := f.KubeClientSet.NetworkingV1().Ingresses(f.Namespace.Name)
+			By("Creating two Ingresses with the issuer name annotation set")
+			_, err := ingClient.Create(context.TODO(), util.NewIngress(ingressName, certificateSecretName, map[string]string{
+				"cert-manager.io/issuer": issuerName,
+			}, acmeIngressDomain), metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			ingress2 := util.NewIngress(ingressName2, certificateSecondSecretName, map[string]string{
+				"cert-manager.io/issuer": issuerName}, acmeIngressDomain)
+			ingress2.Spec.Rules[0].IngressRuleValue.HTTP.Paths[0].Path = "/different"
+
+			_, err = ingClient.Create(context.TODO(), ingress2, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		case util.HasIngresses(f.KubeClientSet.Discovery(), networkingv1beta1.SchemeGroupVersion.String()):
+			ingClient := f.KubeClientSet.NetworkingV1beta1().Ingresses(f.Namespace.Name)
+			By("Creating two Ingresses with the issuer name annotation set")
+			_, err := ingClient.Create(context.TODO(), util.NewV1Beta1Ingress(ingressName, certificateSecretName, map[string]string{
+				"cert-manager.io/issuer": issuerName,
+			}, acmeIngressDomain), metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			_, err = ingClient.Create(context.TODO(), util.NewV1Beta1Ingress(ingressName2, certificateSecondSecretName, map[string]string{
+				"cert-manager.io/issuer": issuerName,
+			}, acmeIngressDomain), metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		default:
+			Fail("Neither " + networkingv1.SchemeGroupVersion.String() + " nor " + networkingv1beta1.SchemeGroupVersion.String() + " were discovered in the API server")
+		}
+
+		By("Waiting for first Certificate to exist")
+		cert, err := f.Helper().WaitForCertificateToExist(ctx, f.Namespace.Name, certificateName, time.Second*60)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for seond Certificate to exist")
+		cert2, err := f.Helper().WaitForCertificateToExist(ctx, f.Namespace.Name, certificateSecondSecretName, time.Second*60)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for the Certificate to be issued...")
+		cert, err = f.Helper().WaitForCertificateReadyAndDoneIssuing(ctx, cert, time.Minute*5)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Waiting for second the Certificate to be issued...")
+		cert2, err = f.Helper().WaitForCertificateReadyAndDoneIssuing(ctx, cert2, time.Minute*5)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Validating the issued Certificate...")
+		err = f.Helper().ValidateCertificate(cert, validations...)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Validating the second issued Certificate...")
+		err = f.Helper().ValidateCertificate(cert2, validations...)
+		Expect(err).NotTo(HaveOccurred())
+	})
```

Comment:
I would like to see an example of a test failure so we are sure the race condition is actually captured.